### ISSUE TITLE
[java] ConsecutiveAppendsShouldReuse false-negative

### ISSUE THREAD

**Issue by jsotuyod** [pmd/pmd#1081]
**Affects PMD Version:** 
6.3.0

**Rule:**
[`java-performance/ConsecutiveAppendsShouldReuse`](https://pmd.github.io/pmd-6.3.0/pmd_rules_java_performance.html#consecutiveappendsshouldreuse)

**Description:**
If the StringBuilder / StringBuffer is not a local (or a parameter after #1068) the violation is not detected.

**Code Sample demonstrating the issue:**

```
public class Foo {
  private final StringBuilder sb = new StringBuilder();

  public String bar() {
    sb.append("foo");
    sb.append("bar");
    sb.append("baz");
    return sb.toString();
  }
}
```

**Running PMD through:** Any

**Comment by jsotuyod** [comment]
No longer reproducible under PMD 7.0.0
